.buildkite/cache-builder.yml

```
@@ -0,0 +1,39 @@
+# Nodes with values to reuse in the pipeline.
+common_params:
+  # Common plugin settings to use with the `plugins` key.
+  - &common_plugins
+    - automattic/bash-cache#add/cocoapods-support: ~
+    - automattic/git-s3-cache#v1.1.0:
+        bucket: "a8c-repo-mirrors"
+        repo: "wordpress-mobile/wordpress-ios/"
+  # Common environment values to use with the `env` key.
+  - &common_env
+    IMAGE_ID: xcode-12.5.1
+
+steps:
+
+  #################
+  # Build the CocoaPods Base Cache
+  #################
+  - label: ":cocoapods: Rebuild CocoaPods cache"
+    command: |
+      echo "--- :rubygems: Setting up Gems"
+      install_gems
+
+      echo "--- :cocoapods: Rebuilding Pod Cache"
+      cache_cocoapods
+    env:
+      IMAGE_ID: xcode-12.5.1
+    <<: *use_plugins
```

Is this working? I don't see an anchor named `&use_plugins` defined anywhere in this YAML, so I believe that would produce a YAML error during parsing…

You probably meant:
```suggestion
    plugins: *common_plugins
```
Since `&common_plugins` is defined as an array of the plugins you want to use here…

---

@AliSoftware is right.

I launched this pipeline manually by passing `PIPELINE=cache-builder.yml` and [it failed](https://buildkite.com/wordpress-mobile/wordpress-ios/builds/1775#93a58005-21d0-4653-b963-c3348ef7ff95):

```
2021-08-02 06:04:44 FATAL  Pipeline parsing of "cache-builder.yml" failed (Failed to parse cache-builder.yml: unknown anchor 'use_plugins' referenced)
```

The reason we didn't notice immediately is that this is not the default pipeline.

In the future, we might benefit from some kind of pipeline validation. So far, I've been validating my changes with 

```
ruby -e "require 'yaml'; puts YAML.load_file('./.buildkite/cache-builder.yml')"
```

---

Addressed in b53bd9e962452335c1d9a9b0101b2e1913ff6090, thanks!